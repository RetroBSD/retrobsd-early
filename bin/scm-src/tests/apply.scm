(apply + '(1 2 3))
(define hypot (lambda (a b) (+ (* a a) (* b b))))
(apply hypot '(3 4))

(for-each (lambda (x y) (display x y) (newline))
    '(q w e r t y) '(1 2 3 4 5 6))
(map (lambda (n) (expt n n)) '(1 2 3 4 5 6))
(map + '(1 2 3) '(4 5 6))
