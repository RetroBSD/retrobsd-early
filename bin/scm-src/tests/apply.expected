(apply + '(1 2 3)) --> 6
(define hypot (lambda (a b) (+ (* a a) (* b b)))) --> <closure-51b6>
(apply hypot '(3 4)) --> 25
q 1
w 2
e 3
r 4
t 5
y 6
(for-each (lambda (x y) (display x y) (newline)) '(q w e r t y) '(1 2 3 4 5 6)) --> ()
(map (lambda (n) (expt n n)) '(1 2 3 4 5 6)) --> (1 4 27 256 3125 46656)
(map + '(1 2 3) '(4 5 6)) --> (5 7 9)
